sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -1091,6 +1092,8 @@ object FoldablePropagation extends Rule[LogicalPlan] {
 object SimplifyCasts extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan.transformAllExpressionsWithPruning(
     _.containsPattern(CAST), ruleId) {
+    case c @ Cast(e: Attribute, StringType, _, _)
```

Comment:
does it have to be Attribute? I think we should match `NamedExpression` so that nested fields can also be covered.